### DISCUSSION TITLE
Tenant specific explore pages

### DISCUSSION THREAD

**Discussion by kameshbhr** [Sunbird-Ed/Community#49] [Bugs] [answered]
Sunbird-Ed Portal supports tenant specific explore pages using "https://<<url>>/<<tenant_id>>/explore". Example in DIKSHA - https://diksha.gov.in/cbse/explore, https://diksha.gov.in/ncert/explore.
The purpose of these tenant specific explore pages is to exclusively show the assets published by them. It is more for a supply-side use case where the tenant who is publishing assets needs a view to see all assets published by them.

We are now able to do this for category specific pages (like Digital Textbooks, Courses etc.) through page configuration where Channel Id (of the tenant) can be included as part of the query parameter. 

But this is working only in the scenario of a guest user (without login). If a logged in user is accessing these pages, it shows assets published by all tenants. There seems to be no way to configure this behaviour. This is an issue, because it is inconsistency.

Irrespective of whether the user is logged in or not, when a user accesses a tenant specific explore page of an asset category, it should show assets that are published only by that tenant. Any further filtering should happen only on top of this.

**Comment by sharathkashyap** [comment]
@kameshbhr Today most of the users of DIKSHA do not have a strong understanding of the concept of tenants. Most of them would access these pages in exploratory purposes. These tenant specific pages are pre-dominantly used by  state department users. As there is an amount of training imparted to such personal. They will be able to identify the difference between normal explore and tenant explore pages.

We are seeing two challenges in this solution:
a) Switch from tenant specific to user preference driven explore pages. We may need to come up with a mechanism where user is aware of the fact that he is inside some tenant specific page and there is a mechanism to switch back to his personal preference driven explore.

b) If a User is already signed in and belongs to specific organisation. His tenant settings are loaded by default. This current solution is solving the issues for most of the tenants today.


This solution needs to be thought through in a more generalized way as I see this problem can be solved for all the users of tenants except NCERT where we do not have a specific framework category term to be identified with.

@luckynvdu I see that this solution needs to be discussed thoroughly as it can impact users in a big way.
cc: @luckynvdu

**Reply by kameshbhr** [reply]
@sharathkashyap I am not sure why your are mixing up the context of DIKSHA with Sunbird-Ed - by thinking about NCERT and other tenants as different. From Sunbird-Ed perspective they should all be same.

Sunbird-Ed is providing a functionality to configure tenant specific explore pages. The expected behaviour of a Tenant specific explore page is to show only the contents published by that Tenant. This functionality is available for guest users (users not logged in). Currently it is achieved through certain configuration. The same functionality should be available for logged in users as well. I am not getting why there is or should be a difference between these users. 
From my understanding , a logged in user also may not belong to a specific tenant (i.e., belong to the default custodian tenant). 
Even if a user belongs to a tenant, if the user is going to a different tenant's explore page, the user should see content that is only published by the tenant to which the explore page belongs to.

Please also do not mix up between a logged in user belonging to a tenant vs. the B, M, G, S configurations in his/her profile. These two should not have any coupling.

This issue is high priority from DIKSHA (open for more than 6 months) and has to be resolved on priority. So request to please take it up asap.

**Reply by pramodkvarma** [reply]
@kameshbhr I am very uncomfortable about the direction we are going with this. Can we have a call next week to discuss this? I would like to understand the thinking behind this. Thanks.

**Reply by kameshbhr** [reply]
Sure @pramodkvarma

**Comment by sharathkashyap** [comment, marked as answer]
@kameshbhr 
 Here is the solution we propose from SunirdED

User has to be onboarded under the organisation as privileged user

Most of the organisations has a framework category term attached. This is the solution that is already in place.

I see the delta in the solution is that organisation can only be associated with framework category term today. It is also possible to extend this to associate any part of content metadata as criteria. This extension being built up will help us to solve the problem holistically for all the use cases.

Ex: Any organisation can start to have association with multiple publishers, multiple content Categories, framework terms etc

This gives the utmost flexibility for any organisation for relevant discovery.

**Comment by sharathkashyap** [comment]
@kameshbhr we are locking this thread from further conversation
